azerothcore/azerothcore-wotlk
# Fix(Core/Stratholme): Some adjustments to Jarien and Sothos fight

## Changes Proposed:
-  Move ReactState to constructor to avoid aggro issues with pets
-  Added mob reference to the talk method to use $n
-  Fixed group loot issues with heirlooms

## Issues Addressed:

- Closes #10258
- Closes #9865 
- Closes https://github.com/chromiecraft/chromiecraft/issues/2830
- Closes https://github.com/chromiecraft/chromiecraft/issues/2829
- Closes https://github.com/chromiecraft/chromiecraft/issues/2650

## SOURCE:


## Tests Performed:

- Tested in-game

## How to Test the Changes:


With 2 chars in party:
Set the loot mode as group loot
.go creature 52149
.additem 6265
summon Void Walker
.additem 22057
use [Brazier of Invocation]
Make sure Jarien and Sothos don't attack Void Walker before conversation ends
kill Sothos/Jarien
watch the other one enrage message
Kill the other one
Loot the chest and you can now roll for the items
